registry/remote/repository.go

```diff
@@ -1288,18 +1300,21 @@ func (s *manifestStore) updateReferrersIndexForPush(ctx context.Context, desc, s
 	return s.repo.delete(ctx, oldIndexDesc, true)
 }
 
-// isReferrersAPIAvailable returns true if the Referrers API is available for r.
-func (r *Repository) isReferrersAPIAvailable(ctx context.Context, desc ocispec.Descriptor) (bool, error) {
+// pingReferrersAPI returns true if the Referrers API is available for r.
+func (r *Repository) pingReferrersAPI(ctx context.Context) (bool, error) {
+	// limit the rate of pinging referrers API
+	r.referrersPingLock.Lock()
+	defer r.referrersPingLock.Unlock()
```

Comment:
Can we add the lock after the line 1314 in order to make it lock free in the most cases?

Reply:
Updated.